emulator.c

```diff
@@ -358,6 +396,14 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0xe6: // NOLINT
       printf("ANI ");
+      immediate = cpu_read_mem(cpu, cpu->pc + 1);
+      cpu->a &= immediate;
+      update_zero_flag(cpu, cpu->a);
+      update_sign_flag(cpu, cpu->a);
+      update_parity_flag(cpu, cpu->a);
+      cpu->flags |= FLAG_CY;
```

Comment:
Does 0xe6 just clear the flag? I think update_carry_flag() etc is clearer.

Reply:
From the i8080 manual page 4-9
The content of the second byte of the instruction is
logicaJly anded with the contents of the accumu lator .
The result is placed in the accumulator. The CY and
AC flags are cleared.

Reply:
using the update function was clearer. looks like i had AC set to be set instead of be cleared so i updated that as well.